### PR TITLE
Use runtime.NumCPU() for num shards and kill worker probability to 0.001

### PR DESCRIPTION
**Pull request by richardartoul**

### CODE REVIEW

File: sync/options.go [hunk 1]
```diff
@@ -21,18 +21,19 @@
...
+	"runtime"
 	"time"
 
 	"github.com/m3db/m3x/instrument"
 )
 
 const (
-	defaultNumShards             = int64(2 ^ 4)
-	defaultKillWorkerProbability = 0.0001
+	defaultKillWorkerProbability = 0.001
```
Comment (arnikola) [inline sync/options.go]: Do we need to edit the default grow kill probability in `config/pooling.go`?

File: sync/options.go [hunk 2]
```diff
@@ -21,18 +21,19 @@
...
 	defaultGrowOnDemand          = false
 )
 
 var (
+	defaultNumShards int64 // Set by init().
```
Comment (justinjc) [inline sync/options.go]: Why not just initialize this to `int64(runtime.NumCPU())` here?

Comment (richardartoul) [reply sync/options.go]: for some reason I thought I couldn't call a function, will fix

File: sync/options.go (hunk 1 above)
Comment (richardartoul) [reply sync/options.go]: good call, done

**Pull request merged**